### PR TITLE
Story/sc 2067 Poetry Upgrade

### PR DESCRIPTION
**Pull request by abnepc**
## Jira Story
[SC-2067](https://epcpower.atlassian.net/browse/SC-2067)
## About
Upgrade `poetry` to `1.5.1` for `m-pm`, `m-bcu`, and `m-tcu`
## Associated Pull Requests

* [m-bcu](https://github.com/epcpower/m-bcu/pull/10)
* [m-tcu](https://github.com/epcpower/m-tcu/pull/59)

## Reproduction Steps
N/A
## Validation
Make sure that `poetry 1.5.1` is being used in CI

[SC-2067]: https://epcpower.atlassian.net/browse/SC-2067?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -91,11 +92,9 @@ exclude = './.venv'
...
 requires = [
-    "alqtendpy",
     "setuptools",
     "wheel",
     "poetry-core>=1.0.0",
-    "poetry-dynamic-versioning",
-
+    "poetry-dynamic-versioning>=1.0.0,<2.0.0",
 ]
 build-backend = "poetry.core.masonry.api"
```
Comment (gordon-epc) [inline pyproject.toml]:
```suggestion
build-backend = "poetry_dynamic_versioning.backend"
```

**Review by gordon-epc — CHANGES_REQUESTED** [review]

File: pyproject.toml (hunk 1 above)
Comment (abnepc) [reply pyproject.toml]: [7b544b4](https://github.com/epcpower/m-pm/pull/6/commits/7b544b4c36437f7c3f6ae595c5eb848501a4b2cd)

File: .github/workflows/ci.yml
Comment (gordon-epc) [inline .github/workflows/ci.yml:1]: Remove lines:
`poetry run poetry-dynamic-versioning`

`poetry run python -c "import sys; print(sys.version)"`

File: README.rst [hunk 2]
```diff
@@ -41,11 +41,11 @@ Windows
 - ``git clone https://github.com/epcpower/m-pm``
 - ``cd m-pm``
 - ``git submodule update --init``
-- ``poetry install``
-- ``poetry run builduimpm``
-- ``poetry run builduiepyqlib``
+- ``poetry@1.5.1 install``
+- ``poetry@1.5.1 run builduipm``
```
Comment (gordon-epc) [inline README.rst]: Be very careful here...
```suggestion
- ``poetry@1.5.1 run builduimpm``
```

File: README.rst [hunk 3]
```diff
@@ -41,11 +41,11 @@ Windows
...
+- ``poetry@1.5.1 run builduiepyqlib``
 
-To launch PM run ``poetry run mpm``.
+To launch PM run ``poetry@1.5.1 run epcpm``.
```
Comment (gordon-epc) [inline README.rst]:
```suggestion
To launch PM run ``poetry@1.5.1 run mpm``.
```

File: README.rst [hunk 4]
```diff
@@ -62,11 +62,11 @@ Linux
 - ``git clone https://github.com/epcpower/m-pm``
 - ``cd m-pm``
 - ``git submodule update --init``
-- ``poetry install``
-- ``poetry run builduimpm``
-- ``poetry run builduiepyqlib``
+- ``poetry@1.5.1 install``
+- ``poetry@1.5.1 run builduipm``
```
Comment (gordon-epc) [inline README.rst]:
```suggestion
- ``poetry@1.5.1 run builduimpm``
```

File: README.rst [hunk 5]
```diff
@@ -62,11 +62,11 @@ Linux
...
+- ``poetry@1.5.1 run builduiepyqlib``
 
-To launch PM run ``poetry run mpm``
+To launch PM run ``poetry@1.5.1 run epcpm``
```
Comment (gordon-epc) [inline README.rst]:
```suggestion
To launch PM run ``poetry@1.5.1 run mpm``
```

File: pyproject.toml
Comment (gordon-epc) [inline pyproject.toml:1]: Remove this line:
`poetry-dynamic-versioning = "^0.13.1"`

**Review by gordon-epc — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml
Comment (abnepc) [reply .github/workflows/ci.yml:1]: [f379261](https://github.com/epcpower/m-pm/pull/6/commits/f379261ae73194947a3749bb1dff88902415ea0a)

File: README.rst (hunk 2 above)
Comment (abnepc) [reply README.rst]: [08d62ce](https://github.com/epcpower/m-pm/pull/6/commits/08d62ced08a29f8e813387fac08480f0519166a0)

File: README.rst (hunk 3 above)
Comment (abnepc) [reply README.rst]: [08d62ce](https://github.com/epcpower/m-pm/pull/6/commits/08d62ced08a29f8e813387fac08480f0519166a0)

File: README.rst (hunk 4 above)
Comment (abnepc) [reply README.rst]: [08d62ce](https://github.com/epcpower/m-pm/pull/6/commits/08d62ced08a29f8e813387fac08480f0519166a0)

File: README.rst (hunk 5 above)
Comment (abnepc) [reply README.rst]: [08d62ce](https://github.com/epcpower/m-pm/pull/6/commits/08d62ced08a29f8e813387fac08480f0519166a0)

File: pyproject.toml
Comment (abnepc) [reply pyproject.toml:1]: [8b31d99](https://github.com/epcpower/m-pm/pull/6/commits/8b31d997a0218d6e3bc37703c22fd2711eb994c8)

**Review by gordon-epc — APPROVED** [review]

**Pull request merged**